lib/core/realtime.rb

```diff
@@ -52,38 +51,134 @@ def run_bot
     end
 
     @client.on :message do |data|
+      # Clean up listners and don't require @bot if in a DM
+      @bot_prefix = "(^#{@bot_name}|^@#{@bot_name}|^\<@#{@client.self.id}\>) " unless data.channel[0] == 'D'
+      @bot_prefix = '^' if data.channel[0] == 'D'
       puts data
-      case data.text
-      when /^#{@bot_name} ping|^@#{@bot_name} ping|^\<@#{@client.self.id}\> ping/ then
-        @client.web_client.chat_postMessage channel: data.channel,
-                                            text: 'PONG'
-      # TODO: Add another listener specific to help command ?
-      # TODO: Decide on how to cache help data to be able to post in chat
-      when /^#{@bot_name} help|^@#{@bot_name} help|^\<@#{@client.self.id}\> help/ then
-        @client.web_client.chat_postMessage channel: data.channel,
-                                            text: 'Imagine a help list here! WHAT?'
-      # TODO: Make output function based on config options (level 1 or 2)
-      # output = @plugins.help data
-      # if output && !output.empty?
-      # TODO: add ability to switch responses from channel or DM based on config settings
-      #  @client.web_client.chat_postMessage channel: data.channel,
-      #                                      text: output
-      when /^#{@bot_name} reload|^@#{@bot_name} reload|^\<@#{@client.self.id}\> reload/ then
-        @client.web_client.chat_postMessage channel: data.channel,
-                                            text: 'Reloading Plugins, please wait.'
-        update_plugin_cache
-        @client.web_client.chat_postMessage channel: data.channel,
-                                            text: 'Plugins Reloaded'
-      # Reads from configuration for bot name or uses the bot name/id from Slack
-      when /^#{@bot_name} |^@#{@bot_name} |^\<@#{@client.self.id}\> / then
-        output = @plugins.exec data
-        if output && !output.empty?
+      if data.user != @client.self.id
+        case data.text
+        when /#{@bot_prefix}ping/ then
```

Comment:
No, In order to allow DM without bot name... I added the space in the `@bot_prefix` variable. It removes the space in the even you are in a DM and just ask `help` or `help plugin`